src/cpp/src/continuous_batching/cache_manager.hpp

```diff
@@ -115,92 +115,101 @@ class CacheManager {
+                            size_t value_roi_size_byte = value_stride + (value_stride & 1) / sub_byte_data_type_multiplier(value_precision);
+                            std::memcpy(reinterpret_cast<uint8_t*>(value_cache.data()), reinterpret_cast<uint8_t*>(m_value_cache[decoder_layer_id].data()), value_roi_size_byte);
+                        } else {
+                            value_roi_size_byte = m_value_cache[decoder_layer_id].get_byte_size();
+                            value_cache_roi_end = static_cast<unsigned char*>(value_cache.data()) + value_roi_size_byte;
+                            ov::Tensor dst_value_roi(value_cache, start_value, end_value);
+                            m_value_cache[decoder_layer_id].copy_to(dst_value_roi);
+                        }
                     }
 
-                    if (value_precision == ov::element::u4) {
-                        size_t value_stride = std::accumulate(end_value.begin(), end_value.end(), 1, std::multiplies<size_t>());
-                        size_t value_roi_size_byte = value_stride + (value_stride & 1) / sub_byte_data_type_multiplier(value_precision);
-                        std::memcpy(reinterpret_cast<uint8_t*>(value_cache.data()), reinterpret_cast<uint8_t*>(m_value_cache[decoder_layer_id].data()), value_roi_size_byte);
+                    // set new cache tensors
+                    if (m_key_cache.size() > decoder_layer_id) {
+                        m_key_cache[decoder_layer_id] = key_cache;
+                        m_value_cache[decoder_layer_id] = value_cache;
                     } else {
-                        value_roi_size_byte = m_value_cache[decoder_layer_id].get_byte_size();
-                        value_cache_roi_end = static_cast<unsigned char*>(value_cache.data()) + value_roi_size_byte;
-                        ov::Tensor dst_value_roi(value_cache, start_value, end_value);
-                        m_value_cache[decoder_layer_id].copy_to(dst_value_roi);
+                        m_key_cache.emplace_back(key_cache);
+                        m_value_cache.emplace_back(value_cache);
                     }
-                }
 
-                // set new cache tensors
-                if (m_key_cache.size() > decoder_layer_id) {
-                    m_key_cache[decoder_layer_id] = key_cache;
-                    m_value_cache[decoder_layer_id] = value_cache;
-                } else {
-                    m_key_cache.emplace_back(key_cache);
-                    m_value_cache.emplace_back(value_cache);
+                    update_request_tensor(decoder_layer_id);
                 }
-
-                update_request_tensor(decoder_layer_id);
+            }
+        }
+        catch (std::exception& e) {
+            if (std::string(e.what()).find("bad allocation") != std::string::npos) {
+                OPENVINO_THROW("Requested KV-cache size is larger than available memory size on the system.");
```

Comment:
Can `std::bad_alloc` be caught instead?

Reply:
I tried to use it, but it didn't work. The type of exception is `ov::Exception`, looks like openvino core wraps `std::bad_alloc` with `ov::Exception` and I couldn't find the API to get the original type, so I changed the type here to `ov::Exception`.